### PR TITLE
Fix daml-sdk-head Maven install

### PR DESCRIPTION
**Pull request by cocreature**
There were two issues:

1. The generated pom.xml used to install everything in one step
referenced the source files in the bazel directory rather than the
released artifacts. This luckily worked so far but falls apart now
since the 2.13 build overwrites the 2.12 build. Same issues that also
blew up the actual release on Thursday.

2. Two artifacts are published as shaded jars. We treated those like
Scala artifacts while they should be treated more like fat
jars. They’re not quite like those since the file paths are different
so I gave them a new release type.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: release/src/Util.hs [hunk 1]
```diff
@@ -81,6 +81,8 @@ data JarType
       -- ^ A java protobuf library (*-speed.jar).
     | Scala
       -- ^ A scala library jar, with source and scaladoc jars.
+    | JarJar
+      -- ^ A scala library jar, with source and scaladoc jars.
```
Comment (garyverhaegen-da) [inline release/src/Util.hs:85]: Should this comment line be slightly different from the previous one? As it stands it makes it a bit difficult to understand why we need a separate type.

Comment (cocreature) [reply release/src/Util.hs]: absolutely that was dumb copypasta c83cea73522203e0b8e52776bff69422fabc054b

**Review by garyverhaegen-da — APPROVED** [review]
I don't understand this diff as well as I'd like, but I can confirm it fixes `daml-sdk-head` for me.

**Comment by cocreature** [comment]
> I don't understand this diff as well as I'd like, but I can confirm it fixes `daml-sdk-head` for me.

Happy to explain anything in here, I definitely did not intend for it to be confusing :slightly_smiling_face:

**Pull request merged**